src/main/java/com/iexec/core/replicate/ReplicatesService.java

```diff
@@ -303,6 +291,53 @@ public void updateReplicateStatus(String chainTaskId,
         }
 
         if (!canUpdate) {
+            return Optional.of(ReplicateStatusUpdateError.GENERIC_CANT_UPDATE);
+        }
+
+        return Optional.empty();
+    }
+
+    /*
+     * We retry up to 100 times in case the task has been modified between
+     * reading and writing it.
+     *
+     * Before updating we check:
+     *   1) if valid transition.
+     *   2) if worker did fail when CONTRIBUTE/REVEAL/UPLOAD_FAILED.
+     *   3) if worker did succeed onChain when CONTRIBUTED/REVEALED.
+     *   4) if worker did upload when RESULT_UPLOADING.
+     */
+    @Retryable(value = {OptimisticLockingFailureException.class}, maxAttempts = 100)
+    public Optional<TaskNotificationType> updateReplicateStatus(String chainTaskId,
+                                                                String walletAddress,
+                                                                ReplicateStatusUpdate statusUpdate) {
+        log.info("Replicate update request [status:{}, chainTaskId:{}, walletAddress:{}, details:{}]",
+                statusUpdate.getStatus(), chainTaskId, walletAddress, statusUpdate.getDetailsWithoutStdout());
+
+        final Optional<ReplicateStatusUpdateError> replicateStatusUpdateError =
+                canUpdateReplicateStatus(chainTaskId,walletAddress,statusUpdate);
+        if (replicateStatusUpdateError.isPresent()
+                && Objects.equals(replicateStatusUpdateError.get(),ReplicateStatusUpdateError.UNKNOWN_REPLICATE)) {
+            log.error(ReplicateStatusUpdateError.UNKNOWN_REPLICATE.getErrorMessageTemplate(), chainTaskId, statusUpdate);
+            return Optional.empty();
+        }
+
+        ReplicatesList replicatesList = getReplicatesList(chainTaskId).orElseThrow();
+        Replicate replicate = replicatesList.getReplicateOfWorker(walletAddress).orElseThrow();
```

Comment:
These throws are not try/catched in the controller.

Reply:
They won't throw anything as they are already checked in `canUpdateReplicateStatus`. IntelliJ triggers the same warning as before when using only `get`.

Reply:
This is an illustration of what I'm trying to say above, `orElseThrow` means possible failure, which in turn means more handling. But I don't know what to do about IntelliJ warnings.